### PR TITLE
Bug fix, Recursion Error #369

### PR DESCRIPTION
**Pull request by SCMusson**
Targeting #369 

TODO:

- [x] implement gt and lt binops for booleans
- [x] implement adding bools and ints
- [x] Write unit tests

### CODE REVIEW

File: opshin/tests/test_misc.py [hunk 1]
```diff
@@ -3024,3 +3024,21 @@ def validator(_: None) -> None:
...
+        self.assertEquals(res, 5 + (a < b))
+
+    @given(a=st.booleans(), b=st.booleans())
+    def test_cast_bool_to_int_gt(self, a: bool, b: bool):
+        source_code = """
+def validator(a: bool, b:bool)-> int:
+    return 5+(a>b)
+"""
+        res = eval_uplc_value(source_code, a, b)
+        self.assertEquals(res, 5 + (a > b))
```
Comment (nielstron) [inline opshin/tests/test_misc.py:3044]:
```suggestion
        self.assertEquals(res, 5 - (a > b))
```

**Comment by nielstron** [comment]
Looks good, thank you @SCMusson 
Please send your Cardano Mainnet Address so that I can transfer the bug bounty!

**Comment by SCMusson** [comment]
addr1qxuyxgyx9qak0nw2pd8wukvy2vdkmgfmr49jkca8ex5nhf493vtsr3yl4f3553sxfx7wzx58cx5pnhekvtlgqf55lw9s3nt0ty

**Comment by nielstron** [comment]
Bounty paid out in e65c681eeadff8283afa7e00c84cb3c3beb6db1db671f0fe57f8e885df956454 🎉

**Pull request merged**